astrobee/survey_manager/scripts/monitor_astrobee

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2021, United States Government, as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# 
+# All rights reserved.
+# 
+# The "ISAAC - Integrated System for Autonomous and Adaptive Caretaking
+# platform" software is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import os
+import socket
+import sys
+import threading
+
+
+def thread_write_input(sock_input):
+    print("starting thread_write_input...")
+    # Store commulative output
+    output_total = ""
+    try:
+        while True:
+            # Get user input dynamically
+            print("get input")
+            user_input = input()
+            print("user input: " + user_input)
+            # Check if the user wants to exit
+            if user_input.lower().strip() == 'exit':
+                break
+            sock_input.send(user_input.encode("utf-8")[:1024])
```

Comment:
user input is always just two characters, anything over that is not valid. So I actually added way more than needed (not sure if that's something to avoid)

Reply:
I'm not saying that 1024 is too short. I'm saying that in general it's bad to accept an invalid input and silently try to sort-of-but-not-really fix it and keep going.

So the recommendation is to switch from silently truncating the string to doing something more sane. It could be reporting a warning and dropping the invalid input, or it could be raising an AssertionError when an assert fails, whatever behavior works out best for the user.